lomond/websocket.py

```diff
@@ -402,19 +409,23 @@ def on_response(self, response):
             )
 
         protocol = response.get('sec-websocket-protocol')
-        extensions = set(response.get_list('sec-websocket-extensions'))
-        self.process_extensions(extensions)
+        extensions = self.process_extensions(
+            response.get_list('sec-websocket-extensions')
+        )
         return protocol, extensions
 
     def process_extensions(self, extensions):
         """Process extension headers."""
+        extensions = set()
```

Comment:
this is a protection against a server which would send repeated headers?

Reply:
It was more to do with how you might use it. The dev would do `my_extension in extensions`, so a set seemed like a logical choice.